Scripts/Runtime/Entities/Util/ComponentSystemBaseExtension.cs

```diff
@@ -143,20 +150,23 @@ public static class ComponentSystemBaseExtension
         /// <param name="system">The system to find the parent of.</param>
         /// <param name="group">The parent of the system</param>
         /// <returns>True if a parent was found for the given system.</returns>
-        public static bool TryFindParentGroup(this ComponentSystem system, out ComponentSystemGroup group)
+        public static bool TryFindParentGroup(this ComponentSystemBase system, out ComponentSystemGroup group)
         {
             var worldSystems = system.World.Systems;
             foreach (ComponentSystemBase worldSystem in worldSystems)
             {
-                if (worldSystem is ComponentSystemGroup worldGroup)
+                if (worldSystem is not ComponentSystemGroup worldGroup)
                 {
-                    Debug.Assert(worldGroup.Systems is List<ComponentSystemBase>);
-                    if ((worldGroup.Systems as List<ComponentSystemBase>).Contains(system))
-                    {
-                        group = worldGroup;
-                        return true;
-                    }
+                    continue;
                 }
+
+                NativeList<SystemHandle> worldGroupSystems = worldGroup.GetAllSystems(Allocator.Temp);
+                if (worldGroupSystems.AsArray().All(systemHandle => systemHandle != system.SystemHandle))
```

Comment:
I don't think we need to introduce `LINQ` here. You could cache the `system.SystemHandle` at the start of the method and then just use `worldGroupSystems.Contains(systemHandle)`.
(Assuming `NativeList` has a contains method.)

Would avoid allocating the conversion to NativeArray too.

Reply:
Good point, done.